packages/amplify-graphql-transformer-core/src/utils/directive-wrapper.ts

```diff
@@ -54,7 +54,7 @@ export class DirectiveWrapper {
       }),
       {},
     );
-    if (options?.deepMergeArguments) {
+    if (options?.deepMergeArguments && needsDeepMerge(defaultValue, argValues)) {
       return _.merge(_.cloneDeep(defaultValue), argValues);
```

Comment:
Do we need 'lodash' here? Can't we do this with object destructuring?

Reply:
There are more than 2 levels in the `defaultValue` that is set by some of the transformers (for example [this](https://github.com/aws-amplify/amplify-category-api/blob/3564898d75c777f6fdf26b8842eed10a179bdf0f/packages/amplify-graphql-index-transformer/src/graphql-index-transformer.ts#L51)) where we set the object and field def GraphQL AST nodes as defaultValues. The `...` spread operator would copy the nested items by reference. This might be okay in majority of cases, but since this is used in multiple transformations, keeping the original deep cloning here. I am not sure how object destructuring is useful here.